src/core/socket.c

```diff
@@ -1224,6 +1224,7 @@ static int usbffs_dispatch_eps(SocketPort *p) {
 
 static int socket_open_fds(Socket *s) {
         _cleanup_(mac_selinux_freep) char *label = NULL;
+        _cleanup_free_ const char *ep;
```

Comment:
This will leave `ep`uninitialized unless the `SOCKET_USB_FUNCTION` case is traveled. The cleanup heandler will pick up the bogus pointer and crash the task.

Please keep the declaration local to `SOCKET_USB_FUNCTION` and add curly braces there.